airbyte-integrations/connectors/source-s3/unit_tests/abstract_test_parser.py

```diff
@@ -2,42 +2,156 @@
 # Copyright (c) 2021 Airbyte, Inc., all rights reserved.
 #
 
-from abc import ABC
+import linecache
+import os
+import random
+import sys
+import tracemalloc
+from abc import ABC, abstractmethod
+from datetime import datetime, timedelta
+from functools import lru_cache, wraps
+from typing import Any, Callable, List, Mapping
 
 import pytest
 from airbyte_cdk import AirbyteLogger
 from smart_open import open as smart_open
+from source_s3.source_files_abstract.file_info import FileInfo
+
+
+def memory_limit(max_memory_in_megabytes: int, print_limit: int = 20) -> Callable:
+    """Runs a test function by a separate process with restricted memory"""
+
+    def decorator(func: Callable) -> Callable:
+        @wraps(func)
+        def wrapper(*args, **kwargs) -> Any:
```

Comment:
I would re-implement this wrapper to achieve the same result and greatly improve speed

```
tracemalloc.start()
result = func(*args, **kwargs)

# get memory usage immediately after function call, we interested in "first_size" value

first_size, first_peak = tracemalloc.get_traced_memory()

# only if we exceeded the quota, build log_messages with traces

if first_size / 1024 ** 2 > max_memory_in_megabytes:
    snapshot = tracemalloc.take_snapshot()
    top_stats = snapshot.statistics("lineno")
    log_messages = .... # build log message
    assert False, log_messages

Reply:
thanks! used